[10.0][ADD] base_external_system_[http|oauth]

NL66278 opened the pull request.

NL66278 committed:
[ADD] base_external_system_oauth: new module

NL66278 committed:
[IMP] base_external_system: refactor

NL66278 committed:
[IMP] base_external_system: better test connection

>A succesfull connection should return true and a failure should
result in an exception. Not the other way around.

NL66278 committed:
[IMP] base_external_system: black

NL66278 committed:
[RFR] base_external_system: correct name for os adapter class

NL66278 committed:
[ADD] base_external_system_http: new module

thomaspaulb commented:
@NL66278 CI is broken and there is talk to phase out python2 odoo versions anyway. I'd say keep this as a private branch and re-propose to OCA upon migration

NL66278 committed:
[IMP] base_external_system: use lambda selection for system_type.

NL66278 committed:
[FIX] base_external_system: load all descendent adapters

NL66278 committed:
[IMP] *_system_http: streamlined logging

NL66278 committed:
[IMP] base..http: improve response error handling

Do not fail when response text contains non ascii chars, and give full response
text back in UserError instead of referring to log.

NL66278 committed:
[IMP] base..oauth: bearer token must be string

For python 2.7 we do want to make sure bearer token contains only valid ascii, and we
should pass token as string to requests.
